integr8ly/keycloak-operator
# feat: add support for provisioning the Keycloak instance once a keycl…

…a CR is created

## Motivation

At the moment, the operator will not actually provision the keycloak instance until a realm is created. In some scenarios, the admin user will want the KC instance to be provision once a KC CR is created.

## How

Add a new "provision" boolean flag to the KC CRD and if it's set to true, the operator will provision the KC instance without waiting for the realm CR.

## Verification Steps

1. Run the operator as normal
2. Create a new keycloak CR with `oc create -f ./deploy/examples/keycloak.json`. You should see that KC is not provisioned as the new flag is not set.
3. Delete the CR `oc delete keycloak example`.
4. Create a new keycloak CR with `oc create -f ./deploy/examples/keycloak_provision.json`. You should see the KC instance is provisioned straight away as the flag is set to true.

## Checklist:
